lib/contact_list_page.dart

```diff
@@ -52,27 +55,35 @@ class _ContactListPageState extends State<ContactListPage> {
         child: ListView(
           padding: EdgeInsets.zero,
           children: <Widget>[
-            const DrawerHeader(
-              decoration: BoxDecoration(
+            DrawerHeader(
+              decoration: const BoxDecoration(
                 color: Colors.blue,
               ),
               child: ListTile(
-                title: Text(
-                  'YanciMan',
-                  style: TextStyle(
-                    fontSize: 16.0,
-                    color: Colors.white,
-                    fontWeight: FontWeight.w400,
-                  ),
-                ),
-                subtitle: Text(
-                  'Producing works of art in Kannywood',
-                  style: TextStyle(
-                    fontSize: 12.0,
-                    color: Colors.white,
-                    fontWeight: FontWeight.w100,
-                  ),
-                ),
+                title: ValueListenableBuilder(
+                    valueListenable: widget.appState.nickName,
+                    builder: (context, nickName, _) {
+                      return Text(
+                        widget.appState.getNick(),
```

Comment:
This should probably be using the `nickName` the builder notifies you of? No need to use (or even have a getter) if we do that. If we drop the `String getX()` methods from the AppState class, we also make sure that no code in the app can accidentally miss setting up a subscription and end up with stale data.